app/src/main/java/org/thoughtcrime/securesms/service/WebRtcCallService.kt

```diff
@@ -724,24 +743,82 @@ class WebRtcCallService : LifecycleService(), CallManager.WebRtcListener {
         }
     }
 
-    private fun setCallInProgressNotification(type: Int, recipient: Recipient?) {
-        try {
-            ServiceCompat.startForeground(
-                this,
-                CallNotificationBuilder.WEBRTC_NOTIFICATION,
-                CallNotificationBuilder.getCallInProgressNotification(this, type, recipient),
-                if (Build.VERSION.SDK_INT >= 30) ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE else 0
+    private fun wakeUpDeviceIfLockedAndDismissKeyguard() {
+        // Get the KeyguardManager and PowerManager
+        val keyguardManager = getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
+        val powerManager = getSystemService(Context.POWER_SERVICE) as PowerManager
+
+        // Check if the phone is locked
+        val isPhoneLocked = keyguardManager.isKeyguardLocked
+
+        // Check if the screen is awake
+        val isScreenAwake = powerManager.isInteractive
+
+        if (!isScreenAwake) {
+            val wakeLock = powerManager.newWakeLock(
+                PowerManager.FULL_WAKE_LOCK or
+                        PowerManager.ACQUIRE_CAUSES_WAKEUP or
+                        PowerManager.ON_AFTER_RELEASE,
+                "${NonTranslatableStringConstants.APP_NAME}:WakeLock"
             )
-        } catch (e: IllegalStateException) {
-            Log.e(TAG, "Failed to setCallInProgressNotification as a foreground service for type: ${type}, trying to update instead", e)
+
+            // Acquire the wake lock to wake up the device
+            wakeLock.acquire(3000)
+
+            while (!powerManager.isInteractive) {
+                /* Busy wait until we're awake - typically this takes less than ~10ms */
+            }
```

Comment:
Is this necessary? What happens if we don't wait for it to be "interactive"? I also worry an accidental deadlock here (what happens if you fail to acquire a wake lock because the system rejects it, etc)

Reply:
It was observed for calls to work slightly better with it, but I've restructured the wake-up call to be a little earlier so now it seems to be about the same without - have removed.